test/domain_bridge/wait_for_publisher.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DOMAIN_BRIDGE__WAIT_FOR_PUBLISHER_HPP_
+#define DOMAIN_BRIDGE__WAIT_FOR_PUBLISHER_HPP_
+
+#include <chrono>
+#include <string>
+#include <thread>
+
+#include "rclcpp/event.hpp"
+#include "rclcpp/node.hpp"
+
+/// Wait for a publisher to be available (or not)
+/*
+ * \return true if the wait was successful, false if a timeout occured.
+ */
+bool wait_for_publisher(
```

Comment:
Functions defined in a header file must be declared `inline`, to avoid breaking ODR.

(though it won't break ODR if we're only including this private header in test files, but just in case I would still add the `inline` specifier)

Reply:
Made inline 60ffdce